docs/core/extensions/snippets/logging/library-authors/DiExampleService.cs

```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.Extensions.Logging;
+
+namespace Logging.LibraryAuthors;
+
+public class DiExampleService
+{
+    private readonly ILogger<DiExampleService> _logger;
+
+    public DiExampleService(ILogger<DiExampleService> logger) =>
+        _logger = logger;
+
+    public void DoSomething()
+    {
+        if (_logger.IsEnabled(LogLevel.Information))
+        {
+            _logger.LogInformation("Doing work...");
```

Comment:
Above we recommended using source-generated logging (👍), so we should follow that recommendation no?

Reply:
Good callout! I'll update this.